src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -106,15 +116,18 @@ private String getParameter(HttpServletRequest request, String name, String defa
     return value == null ? defaultValue : value;
   }
 
-  /** Returns the ID of the current school as an integer. */
-  // TODO:  Add throw exception and display an error message to the end users.
-  private int getId(HttpServletRequest request) {
-    int fail = -1;
-    try {
-      return Integer.parseInt(request.getParameter("id"));
-    } catch (NumberFormatException exception) {
-      System.out.println("getID Invalid parametr: ID request is not a valid number.");
+  /** Returns the nickname of the user with id, or null if the user has not set a nickname. */
+  private String getUserNickname(String userId) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query =
+        new Query("UserInfo")
+            .setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    if (entity == null) {
+      return null;
```

Comment:
I guess we can return a default value like "Anonymous" or so, in case somehow the data messed up?